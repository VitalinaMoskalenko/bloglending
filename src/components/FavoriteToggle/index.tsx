import React, { CSSProperties, useState } from "react";
import styled from "styled-components";
import { H2 } from "../Headings";

type PropsType = {
  style?: CSSProperties;
  isToggle: boolean;
  onClick: () => void;
};

type LikeButtonType = {
  isToggle: boolean;
};

const LikeButton = styled(H2)<LikeButtonType>`
  border: ${({ isToggle, theme }) =>
    isToggle
      ? `1px solid ${theme.colors.red}`
      : `1px solid ${theme.colors.black}`};
  color: ${({ isToggle, theme }) =>
    isToggle ? `${theme.colors.red}` : `${theme.colors.black}`};
  width: 40px;
  height: 40px;
  border-radius: 20px;
  text-align: center;
  cursor: pointer;
  :hover {
    opacity: 0.8;
  }
`;

const FavoriteToggle = ({ isToggle, onClick, style }: PropsType) => {
  return (
    <div>
      <LikeButton onClick={onClick} isToggle={isToggle} style={style}>
        &#10083;
      </LikeButton>
    </div>
  );
};

export default FavoriteToggle;
