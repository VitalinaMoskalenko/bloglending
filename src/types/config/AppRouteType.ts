export enum AppRouteType {
  home = "/",
  postDetail = "/postDetail",
  postDetailsWithId = "/postDetail/:id",
}
