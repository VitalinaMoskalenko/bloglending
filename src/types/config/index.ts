export * from "./FontSizeType";
export * from "./ColorType";
export * from "./AppLocalsType";
export * from "./BreakpointType";
export * from "./AppRouteType";
