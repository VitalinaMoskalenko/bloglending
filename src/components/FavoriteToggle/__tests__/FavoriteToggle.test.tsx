import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import FavoriteToggle from "../";
import { theme } from "../../../config";

it("renders FavoriteToggle correctly", () => {
  const favoriteToggle = renderer
    .create(
      <ThemeProvider theme={theme}>
        <FavoriteToggle isToggle={true} onClick={() => {}} />
      </ThemeProvider>
    )
    .toJSON();

  expect(favoriteToggle).toMatchSnapshot();
});
