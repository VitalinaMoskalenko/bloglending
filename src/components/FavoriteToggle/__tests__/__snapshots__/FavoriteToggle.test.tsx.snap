// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FavoriteToggle correctly 1`] = `
<div>
  <h1
    className="sc-gtsrHT sc-jSFjdj kMImv bhGncE"
    onClick={[Function]}
  >
    ‚ù£
  </h1>
</div>
`;
